### DISCUSSION TITLE
Kbar and wildcard search?

### DISCUSSION THREAD

**Discussion by dariusz-wozniak** [timlrx/tailwind-nextjs-starter-blog#767] [Q&A]
I have a problem with the wildcard search:

Example (reproducible on [my blog]):

Main window:

> ![image](https://github.com/timlrx/tailwind-nextjs-starter-blog/assets/5969160/c03f1551-c846-47c4-801a-f803fd424d6d)

Typing `Autom` yields a result [OK]:

> ![image](https://github.com/timlrx/tailwind-nextjs-starter-blog/assets/5969160/e61dd9a1-462f-4786-9451-d57c7e9284e9)

While typing `regi` (as for _register_) does not yield any results [NOT OK]:

> ![image](https://github.com/timlrx/tailwind-nextjs-starter-blog/assets/5969160/19c15b31-60ee-429a-bcd1-10e645a5c960)

How can I make `register` to find the proper page?

**Comment by timlrx** [comment]
Kbar uses fuse.js internally for the searching - https://github.com/timc1/kbar/blob/main/src/useMatches.tsx#L12-L29

Fuse.js does [approximate string match](https://www.fusejs.io/concepts/scoring-theory.html). I noticed that the default settings don't work quite well for long strings. They get penalize by the location distance and Field-length Norm. You can try modifying one of the entries in fuse demo to use your blog's title - using "regi" as a search term ranks very low.

The best solution would be for kbar to improve their settings or expose an option to customize fuse.js parameters.
